### DISCUSSION TITLE
Manipulate YamlNode

### DISCUSSION THREAD

**Discussion by emartynov** [charleskorn/kaml#384] [Q&A] [answered]
Hi people,

I'm writing a small kotlin script to read yaml files and translate English scalar to another language.
The structure of files is different and I don't want to deserialize it to some structure but I want to iterate the nodes in the tree, and if it is scalar, then add content for the translation.

I can not grasp how could I deserialize the modified YamlMap to string?

**Comment by emartynov** [comment]
Looking at the code, I probably have to iterate over the tree and write yaml file by hand.

**Comment by charleskorn** [comment, marked as answer]
Kaml is not intended to be a general-purpose YAML manipulation library. I'd suggest using something like SnakeYAML for this purpose instead.

**Reply by emartynov** [reply]
Thank you so much for the answer and library! I learned a lot while toying with it and finally, I can read structures from the YAML files.
